corgi/tasks/manifest.py

```diff
@@ -16,31 +19,69 @@
     retry_kwargs=RETRY_KWARGS,
 )
 def update_manifests():
+    external_names = set()
     for ps in ProductStream.objects.annotate(num_components=Count("components")).filter(
         num_components__gt=0
     ):
-        cpu_update_ps_manifest.delay(ps.name)
+        # Don't regenerate a manifests for streams with matching external names, the content will
+        # be the same. This happens for the following streams, which share the same brew_tags, and
+        # variants
+        # CERTSYS-10.4-RHEL-8: ['certificate_system_10.4', 'certificate_system_10.4.z']
```

Comment:
For certificate_system 10.4, and 10.4.z the SBOM contents match as expected in stage. For some reason in prod they are different, however the 2 root components are the same, so perhaps some ordering is different or similar.

```
>>> import corgi.core.models as m
>>> cert104 = m.ProductStream.objects.get(name="certificate_system_10.4")
>>> cert104.components.root_components().released_components().values("purl")
<ComponentQuerySet [{'purl': 'pkg:rpmmod/redhat/redhat-pki@10:8060020220420152504:07fb4edf'}, {'purl': 'pkg:rpmmod/redhat/redhat-pki@10:8060020220824122647:07fb4edf'}]>
>>> cert104z = m.ProductStream.objects.get(name="certificate_system_10.4.z")
>>> cert104z.components.root_components().released_components().values("purl")
<ComponentQuerySet [{'purl': 'pkg:rpmmod/redhat/redhat-pki@10:8060020220420152504:07fb4edf'}, {'purl': 'pkg:rpmmod/redhat/redhat-pki@10:8060020220824122647:07fb4edf'}]>
```

I guess it is possible for 2 streams to have matching external_names but have different yum_repos, or composes. However in practice that does not occur in the data we have in product_definitions today.

Once we release this change to production we will need to remove all the current files, regenerate and publish new ones.

Reply:
The certificate-system-10.4 stream doesn't have any latest_components in prod. It's strange because both have the same set of root components. However I don't think it's worth spending a great deal of time investigating it as the streams are the same in prod. We should check the 2 streams again after we do our next prod deploy.

```
>>> stream = m.ProductStream.objects.get(name="certificate_system_10.4.z")
>>> stream.components.root_components().released_components().values_list("purl",flat=True)
<ComponentQuerySet ['pkg:rpmmod/redhat/redhat-pki@10:8060020220420152504:07fb4edf', 'pkg:rpmmod/redhat/redhat-pki@10:8060020220824122647:07fb4edf']>
>>> stream.components.root_components().released_components().latest_components(ofuri=stream.ofuri,include_inactive_streams=True).values_list("purl",flat=True)
<ComponentQuerySet ['pkg:rpmmod/redhat/redhat-pki@10:8060020220824122647:07fb4edf']>
```

```
>>> stream = m.ProductStream.objects.get(name="certificate_system_10.4")
>>> stream.components.root_components().released_components().values_list("purl",flat=True)
<ComponentQuerySet ['pkg:rpmmod/redhat/redhat-pki@10:8060020220420152504:07fb4edf', 'pkg:rpmmod/redhat/redhat-pki@10:8060020220824122647:07fb4edf']>
>>> stream.components.root_components().released_components().latest_components(ofuri=stream.ofuri,include_inactive_streams=True).values_list("purl",flat=True)
<ComponentQuerySet []>
```

I think the probably the `include_inactive_streams` setting is not being passed through to the stored procedure in prod. Although that part of the code in production looks OK to me.

Reply:
I filed an new issue for this: CORGI-981